packages/expo-updates/android/src/main/java/expo/modules/updates/loader/FileDownloader.kt

```diff
@@ -319,6 +333,12 @@ open class FileDownloader(context: Context) {
             }
 
             override fun onSuccess(file: File, hash: ByteArray) {
+              val hashHexString = String(Hex.encodeHex(hash))
+              if (asset.expectedHash != null && !asset.expectedHash.equals(hashHexString)) {
```

Comment:
```suggestion
              if (asset.expectedHash != null && asset.expectedHash != hashHexString) {
```

Also be careful with differences in case. We don't specify the case (or even hex encoding) of the hash in our spec and need to do that. (Ex: maybe allow both upper and lowercase in the manifest, but need to normalize the case in client code here.)

Reply:
https://linear.app/expo/issue/ENG-2695/add-hash-casing-to-spec

Reply:
https://github.com/expo/expo/pull/15555